mysql/README.md

```diff
@@ -181,6 +181,12 @@ See our [sample mysql.yaml][9] for all available configuration options, includin
             source: mysql
             sourcecategory: database
             service: myapplication
+            log_processing_rules:
+              - type: multi_line
+                # Depending on your MySQL configuration, slow query logs may start with a
+                # different line, so you may need to tweak this pattern.
+                name: new_log_starts_with_time
+                pattern: "# Time:"
```

Comment:
What do you mean `slow query logs may start with a different line`?

Like this `# Query_time` ? https://github.com/DataDog/logs-backend/pull/6293/files#diff-eef4481fd2beac05fbede10a33a29f63R37

Reply:
If that's the case, maybe we can suggest all cases `pattern: "# (Time|Query_time|Something_else):"`

Reply:
Yes, e.g. the `short_log_format` option on `mysqld` may result in skipping the first two lines, making a slow log entry start with a `# Query_time` line.

I'll look at what I can do with that approach!

Reply:
I thought about it, and I *think* this can't work, because we'd end up splitting slow log entries twice (on `# Time` and `# Query_time`) -- details below.

We *can*, however, suggest all possible known cases to our users. In practice there are only two possible cases, although the vast majority of users will just need `# Time:`:

```yaml
pattern: "# Time:"
# If mysqld was started with `--log-short-format`, use:
# pattern: "# Query_time:"
```

---

<details>
<summary>
Details (click to expand)
</summary>

So, it's a bit tricky. There are two possible cases: starts with `# Time`, or starts with `# Query_time`:

- Full format (MySQL) - starts with `# Time`:

```
# Time: 2019-11-27T10:43:13.460744Z
# User@Host: root[root] @ localhost [1.2.3.4]  Id:    35
# Query_time: 0.214922  Lock_time: 0.000184 Rows_sent: 248832  Rows_examined: 72
...
```

- Short format (MySQL) - starts with `# Query_time`:

```
# Query_time: 0.214922  Lock_time: 0.000184 Rows_sent: 248832  Rows_examined: 72
...
```

- Full format (MariaDB) - starts with `# Time`:

```
# Time: 191127 15:45:44
# User@Host: root[root] @ localhost [1.2.3.4]
# Thread_id: 55  Schema: app  QC_hit: No
# Query_time: 0.214922  Lock_time: 0.000159 Rows_sent: 248832  Rows_examined: 12
...
```

We can't use a union regex (e.g. `# (Time|Query_time)`) because the Agent might wrongly split groups of log lines. For example for MariaDB, we'd end up with:

- Split on `# Time`:

```
# Time: 191127 15:45:44
# User@Host: root[root] @ localhost [1.2.3.4]
# Thread_id: 55  Schema: app  QC_hit: No
```

- Split on `# Query_time`:

```
# Query_time: 0.214922  Lock_time: 0.000159 Rows_sent: 248832  Rows_examined: 12
# Rows_affected: 0
use app;
SET timestamp=1574869544;
SELECT * FROM fruit f1, fruit f2, fruit f3, fruit f4, fruit f5;
```

So the user needs to configure only *one* of the patterns, and which pattern of `# Time` or `# Query_time` depends on their configuration. More specifically, whether they're using MySQL with [`--log-short-format`](https://dev.mysql.com/doc/refman/8.0/en/server-options.html#option_mysqld_log-short-format) enabled.
</details>

Reply:
👍, Thanks for the explanation